src/Allure.XUnit/AllureXunitHelper.cs

```diff
@@ -115,17 +117,32 @@ internal static void Log(string message)
             AllureMessageBus.TestOutputHelper.Value.WriteLine("╬═══════════════");
         }
 
+        private static void AddDistinct(this List<Label> labels, Label labelToInsert)
+        {
+            var existingLabel = labels.FirstOrDefault(label => label.name.Equals(labelToInsert.name));
+            if (existingLabel != null)
+            {
+                existingLabel.value = labelToInsert.value;
+            }
+            else
+            {
+                labels.Add(labelToInsert);
+            }
+        }
+
         private static void UpdateTestDataFromAttributes(TestResult testResult, ITestCase testCase)
         {
-            var attributes = testCase.TestMethod.Method.GetCustomAttributes(typeof(IAllureInfo));
+            var classAttributes = testCase.TestMethod.TestClass.Class.GetCustomAttributes(typeof(IAllureInfo));
```

Comment:
I think, that doesn't obviously what kind of attributes overwrite value  attributes same types what not.
I propose make that all attributes overwrite value same types attributes  and let's make AllowMultiple=false for all, but allow pass multiple values by 'params' for some attributes which it's need. What do you think?

Reply:
Vote up for disallowing multiple usage of attributes.

Overwriting values for every attribute could make unusable AllureTagAttribute for class and method same time.
ex.:
```
    [AllureTag("global_tag_1", "global_tag_2")]
    public class SomeTestClass
    {
        [AllureXunit]
        [AllureTag("local_tag")]
        public void TestMethod()
        {
            // some test stuff
        }
    }
```
Same thing for AllureLinkAttribute.

Possible solution: Adding "Overwrite" boolean parameter for allure attributes in the end of definition
Pros:
- Simple and understandable way to overwrite|append values for any kind of allure labels
Cons:
- A bit more complex way to pass multiple values for labels, we cannot use `params string[] args` anymore.
ex.:
```
[AllureTag(new[] {"TAG-8", "TAG-9", "TAG-10"})]
// or
[AllureTag(new[] {"TAG-8", "TAG-9", "TAG-10"}), true]
```

Your turn :)